src/base/map_util.h

```diff
@@ -0,0 +1,96 @@
+/*********************                                                        */
+/*! \file map_util.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Tim King
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Utility functions for dealing with maps in a mostly uniform fashion.
+ **
+ ** Utility functions for dealing with maps and related classed in a mostly
+ ** uniform fashion. These are stylistically encouraged (but not required) in
+ ** new code. Supports:
+ ** - std::map
+ ** - std::unordered_map
+ ** - CVC4::context::CDHashmap
+ ** - CVC4::context::CDInsertHashmap
+ **
+ ** Currently implemented classes of functions:
+ ** - ContainsKey
+ **   Returns true if a map contains a key. (Also Supports std::set and
+ **   std::unordered_set.)
+ ** - FindOr*
+ **   Finds an data element mapped to by the map. Variants include FindOrNull
+ **   and FindOrDie.
+ **
+ ** Potential future classes of functions:
+ ** - InsertOrUpdate
+ ** - InsertIfNotPresent
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef __CVC4__MAP_UTIL_H
+#define __CVC4__MAP_UTIL_H
+
+#include "base/cvc4_check.h"
+
+namespace CVC4 {
+
+// Returns true if the `map` contains the `key`.
+//
+// Supports sets as well.
+template <class M, class Key>
+bool ContainsKey(const M& map, const Key& key)
```

Comment:
Nit: I think that we are lacking an explicit guideline on this right now but we generally use lower camelCase for function names.

Reply:
The style guide is consistent with using mixed case in examples for functions: [FooFactory and FooConsumer](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#ownership-and-smart-pointers) and [Function1 and Function2](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#nonmember-static-member-and-global-functions). FWIW the [google style guide](https://google.github.io/styleguide/cppguide.html#Function_Names) suggests mixed case for functions. My personal preference also goes against this for free functions. (Methods are more ambiguous and I think this is what you are talking about.) In absence of a guideline to the contrary, I think we should just move forward with this as is. If you think renaming this is an improvement, I recommend doing this in a follow up.